packages/frontend/core/src/modules/media/entities/audio-transcription-job.ts

```diff
@@ -0,0 +1,357 @@
+    } catch (err) {
+      logger.debug('Error during job submission', { error: err });
+      this._status$.value = {
+        status: AiJobStatus.failed,
+        error: UserFriendlyError.fromAny(err),
+      };
+    }
+    return this.status$.value;
+  }
+
+  private async untilJobFinishedOrClaimed() {
+    while (
+      !this.disposed &&
+      this.props.blockProps.jobId &&
+      this.props.blockProps.createdBy === this.currentUserId
+    ) {
+      logger.debug('Polling job status', {
+        jobId: this.props.blockProps.jobId,
+      });
+      const job = await this.fetchers.getAudioTranscription(
+        this.props.blobId,
+        this.props.blockProps.jobId
+      );
+
+      if (!job || job?.status === 'failed') {
+        logger.debug('Job failed during polling', {
+          jobId: this.props.blockProps.jobId,
+        });
+        throw UserFriendlyError.fromAny('Transcription job failed');
+      }
+
+      if (job?.status === 'finished' || job?.status === 'claimed') {
+        logger.debug('Job finished, ready to claim', {
+          jobId: this.props.blockProps.jobId,
+        });
+        this._status$.value = {
+          status: AiJobStatus.finished,
+        };
+        return;
+      }
+
+      // Add delay between polling attempts
+      await new Promise(resolve => setTimeout(resolve, 3000));
+    }
+  }
+
+  async claim() {
+    if (this.disposed) {
+      logger.debug('Job already disposed, cannot claim');
+      throw new Error('Job already disposed');
+    }
+
+    logger.debug('Attempting to claim job', {
+      jobId: this.props.blockProps.jobId,
+    });
+
+    if (!this.props.blockProps.jobId) {
+      logger.debug('No job id found, cannot claim');
+      throw new Error('No job id found');
+    }
+
+    const claimedJob = await this.fetchers.claimAudioTranscription(
+      this.props.blockProps.jobId
+    );
+
+    if (claimedJob) {
+      logger.debug('Successfully claimed job', {
+        jobId: this.props.blockProps.jobId,
+      });
+      const result: TranscriptionResult = {
+        summary: claimedJob.summary ?? '',
+        title: claimedJob.title ?? '',
+        segments:
+          claimedJob.transcription?.map(segment => ({
+            speaker: segment.speaker,
+            start: segment.start,
+            end: segment.end,
+            transcription: segment.transcription,
+          })) ?? [],
+      };
+
+      this._status$.value = {
+        status: AiJobStatus.claimed,
+        result,
+      };
+    } else {
+      throw new Error('Failed to claim transcription result');
+    }
+  }
+
+  isCreator() {
+    return (
+      this.props.blockProps.jobId &&
+      this.props.blockProps.createdBy &&
+      this.props.blockProps.createdBy === this.currentUserId
+    );
+  }
+
+  private get serverService() {
+    // server scope is not available in the entity
+    // here is a workaround to get the server service
+    const serverService = this.framework.get(ServersService);
```

Comment:
how about `WorkspaceServerService.server`?

If this also works in the local workspace, you should use `WorkspaceServerService.server || DefaultServerService.server`